nntrainer/layers/conv2d_layer.cpp

```diff
@@ -102,6 +110,33 @@ static void col2im(const Tensor &col_matrix, const TensorDim &kdim,
   }
 }
 
+static TensorDim
+calcIm2ColOutputDim(const TensorDim &in, const TensorDim &kdim,
+                    const std::array<unsigned int, CONV2D_DIM> &padding,
+                    const std::array<unsigned int, CONV2D_DIM> &mstride,
+                    const std::array<unsigned int, CONV2D_DIM> &dilation) {
+
+  unsigned int ph = padding[0];
+  unsigned int pw = padding[1];
+
+  int in_height = in.height();
+  int in_width = in.width();
+  unsigned int height = in_height + ph * 2;
+  unsigned int width = in_width + pw * 2;
+  unsigned int k_height = kdim.height();
+  unsigned int k_width = kdim.width();
+
```

Comment:
I guess, @zhoonit update the padding calculation. Just check if it is valid.

Reply:
This will have to change, I'll make sure this remains valid when reviewing v2 -> main 👍